cylc/flow/scheduler.py

```diff
@@ -1560,14 +1569,21 @@ def run(self):
             self.main_loop_intervals.append(time() - tinit)
             # END MAIN LOOP
 
-    def update_state_summary(self):
-        """Update state summary."""
+    def update_data_structure(self):
```

Comment:
I think we can keep the previous function name, and update only its docs?

Reply:
? I change it because it wasn't only updating the sate summary (aka 'state_summary_mgr.py') but also the DB and now UI Feed... IMO data structure represents an instantiation of the data model, so I thought it more appropriate now as it can encapsulate all three..

Do you have any reason for the name `update_state_summary`?

Reply:
>Do you have any reason for the name update_state_summary?

Not really. The previous name could be misleading or incomplete, as it was updating data in the `state_summary_mgr`, updating the database in `suite_db_mgr`, and even updating the `Scheduler` state fields. But I assume the previous developer either considered all that part of the state summary. Otherwise, we could have a separate ticket to rename this method, as I *think* if it was an existing problem with the method name, it doesn't necessarily need to be fixed in the GraphQL change.

But just my opinion, if others prefer to rename the method here all good for me :+1:

Reply:
(And I actually prefer the name you used, though I am not sure what's the difference between state summary and data structure, as it is updating more things than just what is in the state summary it may well be more logical to have this name)

Reply:
state summary was/is a data extraction as a dump for the old GUI.

This all might change with incremental updates (if done in this PR), but for now I needed to update the protobuf messages using the same logic.. So don't want it to be called under a misleading name.

Reply:
Oh, got it. Sorry for the lack of knowledge about the previous GUI. I created some other tickets for things that were legacy too, just to have to close the ticket. At least I learned about some background history. Thanks @dwsutherland ! Resolving this one :+1: